src/optionals/system.c

```diff
@@ -100,6 +100,36 @@ static Value chownNative(DictuVM *vm, int argCount, Value *args) {
 
     return newResultSuccess(vm, EMPTY_VAL);
 }
+
+static Value mkdirTempNative(DictuVM *vm, int argCount, Value *args) {
+    if (argCount == 0) {
+        runtimeError(vm, "mkdirTemp() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_STRING(args[0])) {
+        runtimeError(vm, "mkdirTemp() first argument must be a string");
+        return EMPTY_VAL;
+    }
+
+    char *template = AS_CSTRING(args[0]);
+    char *tmpl = {0};
+
+    if (template[0] != '\0') {
+        tmpl = malloc(strlen(template)+1);
```

Comment:
I try to avoid uses of malloc / alloc etc where possible and try to use the memory macros, this means if in the future we ever needed to change the memory allocator it's a simple switch.

We'll need to ensure we're freeing things up too since on success where creating a new string rather than inheriting the already alloc'd one.